### PR TITLE
allow negative numbers in free opts

### PR DESCRIPTION
**Pull request by slyshykO**

### CODE REVIEW

**Comment by RazrFalcon** [comment]
This is indeed an interesting edge case.

Could you keep the original formatting and also add some tests? Thanks.

File: src/lib.rs [hunk 1]
```diff
@@ -693,7 +695,7 @@ impl Arguments {
         let mut flags_left = Vec::new();
         for arg in &self.0 {
             if let Some(s) = arg.to_str() {
-                if s.starts_with('-') && s != "-" {
+                if s.starts_with('-') && s != "-" && !s.chars().nth(1).unwrap_or('f').is_numeric() {
```
Comment (RazrFalcon) [inline src/lib.rs:698]: You can replace this with `s.chars().skip(1).all(|c| c.is_ascii_digit() || c == '.'))`

**Comment by RazrFalcon** [comment]
Thanks.
